src/main/scala/com/codacy/rules/ConfigurationRules.scala

```diff
@@ -65,38 +70,78 @@ class ConfigurationRules(cmdConfig: CommandConfiguration) extends StrictLogging
-            val error = s"Invalid CODACY_API_BASE_URL: ${config.codacyApiBaseUrl}"
-
-            val help = if (!config.codacyApiBaseUrl.startsWith("http")) {
-              "Maybe you forgot the http:// or https:// ?"
-            }
-            Left(s"$error\n$help")
-
-          case config if config.projectToken.trim.isEmpty =>
-            Left("Empty argument for --project-token")
-
-          case _ =>
-            Right(baseConf)
-        }
-      }
+      validatedConfig <- validateBaseConfigUrl(baseConf)
     } yield {
-      logger.info(s"Using API base URL: ${validatedConfig.codacyApiBaseUrl}")
+      logger.info(s"API base URL: ${validatedConfig.codacyApiBaseUrl}")
       validatedConfig
     }
   }
 
+  private def validateAuthConfig(
+      baseCommandConfig: BaseCommandConfig,
+      envVars: Map[String, String]
+  ): Either[String, AuthenticationConfig] = {
+    val errorMessage =
+      "Either a project token or an api token must be provided or available in an environment variable"
+
+    val projectToken = getValueOrEnvironmentVar(baseCommandConfig.projectToken, envVars, "CODACY_PROJECT_TOKEN")
+    val apiToken = getValueOrEnvironmentVar(baseCommandConfig.apiToken, envVars, "CODACY_API_TOKEN")
+
+    if (projectToken.isDefined)
+      validateProjectTokenAuth(projectToken)
+    else if (apiToken.isDefined)
+      validateApiTokenAuth(baseCommandConfig, apiToken, envVars)
+    else
+      Left(errorMessage)
+  }
+
+  private def validateProjectTokenAuth(projectToken: Option[String]) =
+    projectToken.filter(_.nonEmpty) match {
+      case None => Left("Empty argument for --project-token")
+      case Some(projectToken) => Right(ProjectTokenAuthenticationConfig(projectToken))
+    }
+
+  private def validateApiTokenAuth(
+      baseCommandConfig: BaseCommandConfig,
+      apiToken: Option[String],
+      envVars: Map[String, String]
+  ) =
+    for {
+      apiToken <- apiToken.filter(_.nonEmpty).toRight("Empty argument --api-token")
+      username <- getValueOrEnvironmentVar(baseCommandConfig.username, envVars, "CODACY_USERNAME")
+        .filter(_.nonEmpty)
+        .toRight("Empty argument --username")
+      projectName <- getValueOrEnvironmentVar(baseCommandConfig.projectName, envVars, "CODACY_PROJECT_NAME")
+        .filter(_.nonEmpty)
+        .toRight("Empty argument --project-name")
+    } yield ApiTokenAuthenticationConfig(apiToken, username, projectName)
+
+  private def getValueOrEnvironmentVar(value: Option[String], envVars: Map[String, String], envVarName: String) =
+    value.orElse(envVars.get(envVarName))
+
+  private def validateBaseConfigUrl(baseConfig: BaseConfig) = baseConfig match {
+    case config if !validUrl(config.codacyApiBaseUrl) =>
+      val error = s"Invalid CODACY_API_BASE_URL: ${config.codacyApiBaseUrl}"
+
+      val help = if (!config.codacyApiBaseUrl.startsWith("http")) {
+        "Maybe you forgot the http:// or https:// ?"
+      }
+      Left(s"$error\n$help")
```

Comment:
Could you use multiline strings?
```suggestion
      Left(s"""$error
              |$help""".stripMargin)
```